### PR TITLE
Loadcell

### PR DESCRIPTION
**Pull request by ryunkim00**

### CODE REVIEW

File: src/HermesController.cpp [hunk 1]
```diff
@@ -51,7 +52,11 @@ bool HermesController::init(void)
 	status &= registerObserver(&mButtonPressPinWatcher);
 	registerMessageHandler(&mChargeRdyMsgHandler);
 	registerMessageHandler(&mBluetoothStatusMsg);
+	registerTimer(&mCheckStateTimer);
+
+	mCheckStateTimer.start(250, PERIODIC);
```
Comment (definededits) [inline src/HermesController.cpp]: maybe only start this once we change into LOAD_SENSOR mode

File: src/HermesController.cpp [hunk 2]
```diff
@@ -425,4 +430,26 @@ void HermesController::onButtonPress(Pin_t pin, int16_t state)
...
+	hw::LoadSensor::LeanState_t leanState = mLoadSensor.getLeanState();
+
+	if (mControl == LOAD_SENSOR) {
+		if (leanState == hw::LoadSensor::FORWARD) {
+			mMotorController.setThrottleInput(100);
+		}
+	}
+}
+
+void HermesController::onTimerExpire(uint32_t userdata)
```
Comment (definededits) [inline src/HermesController.cpp:457]: duplicate code???

File: src/hw/MotorInterface.cpp [hunk 3]
```diff
@@ -116,16 +89,19 @@ MaxAccel_t MotorInterface::getMaxAccel(void) const
 void MotorInterface::motorOn(void)
 {
 	mMotorOn = true;
-	//if (!mWheelSpeedTimer.isEnabled()) {
-	//	mWheelSpeedTimer.start(100, PERIODIC);
-	//}
+	if (!mThrottleUpdateTimer.isEnabled()) {
+		mThrottleUpdateTimer.startMicros(19600, PERIODIC);
+	if (!mThrottleUpdateTimer.isEnabled()) {
```
Comment (definededits) [inline src/hw/MotorInterface.cpp]: ??

File: src/hw/MotorInterface.cpp [hunk 4]
```diff
@@ -116,16 +89,19 @@ MaxAccel_t MotorInterface::getMaxAccel(void) const
...
+	}
 	return;
 }
 
 void MotorInterface::motorOff(void)
 {
 	mMotorOn = false;
-	//mWheelSpeedTimer.stop();
+	mThrottleUpdateTimer.stop();
+	mThrottleUpdateTimer.stop();
```
Comment (definededits) [inline src/hw/MotorInterface.cpp]: ??

File: src/hw/MotorInterface.cpp [hunk 5]
```diff
@@ -152,22 +128,25 @@ uint8_t MotorInterface::getSpeedKmh(void)
...
 {
-	//uint16_t h1 = analogRead(MOTOR_SENSE_H1);
-	//uint16_t h2 = analogRead(MOTOR_SENSE_H2);
-	//uint16_t h3 = analogRead(MOTOR_SENSE_H3);
+	if (mThrottle > mPwmSignal) {
+		mPwmSignal++;
+	} else if (mThrottle < mPwmSignal) {
+		mPwmSignal -= 10;
+	}
+
```
Comment (definededits) [inline src/hw/MotorInterface.cpp:133]: ????????

Comment (ryunkim00) [reply src/hw/MotorInterface.cpp:133]: The plot thickens

**Review by definededits — APPROVED** [review]

**Pull request merged**